lib/config.py

```diff
@@ -294,14 +295,22 @@ def parse(self, command_line_args=None):
             for key, value in command_line_args.items():
                 if key in config[node_name]:
                     if value is not None:
-                        config[node_name][key] = value
+                        value_type = type(config[node_name][key])
+                        if  value_type == str or value_type == bool or value_type == list:
+                            config[node_name][key] = value
+                        elif type(config[node_name][key]) == dict:
+                            config[node_name][key] = json.loads(value)
                     command_line_args.pop(key)
         # update common config node with remaining command line arguments
         node_name = "common"
         for key, value in command_line_args.items():
             if key in config[node_name]:
                 if value is not None:
-                    config[node_name][key] = value
+                    value_type = type(config[node_name][key])
+                    if value_type == str or value_type == bool or value_type == list:
+                        config[node_name][key] = value
+                    elif type(config[node_name][key]) == dict:
+                        config[node_name][key] = json.loads(value)
                 command_line_args.pop(key)
```

Comment:
I agree in having the function, however it needs some modifications to actually modify the dict:
```python
def assign_config_value(config_dict, key, value):
  value_type = type(config_dict[key])
  if value_type == dict:
    config_dict[key] = json.loads(value)
  else:
    config_dict[key] = value
```